src/System.Threading.Overlapped/tests/ThreadPoolBoundHandle_AllocateNativeOverlappedTests.cs

```diff
@@ -83,7 +83,9 @@ public unsafe void AllocateNativeOverlapped_NonBlittableTypeAsPinData_Throws()
     {
         using(ThreadPoolBoundHandle handle = CreateThreadPoolBoundHandle())
         {
-            AssertExtensions.Throws<ArgumentException>(null, () => handle.AllocateNativeOverlapped((_, __, ___) => { }, new object(), new NonBlittableType() { s = "foo" }));
+            AssertExtensions.Throws<ArgumentException>(
+                PlatformDetection.IsFullFramework ? null : "userData",
+                () => handle.AllocateNativeOverlapped((_, __, ___) => { }, new object(), new NonBlittableType() { s = "foo" }));
```

Comment:
```C#
AssertExtensions.Throws<ArgumentException>(
    "userData",
    null,
    () => ...);
```

Is close to this same behavior, but for that method `null` to netfx means "don't check" instead of "check it's actually null".

If checking that it's actually null is useful, it's probably worth a comment saying that using the overload is wrong; otherwise consider using the overload.

Reply:
@bartonjs Thanks, I figued out that the method exists based on the feedback on another PR. Would have been the perfect solution here.